tls/s2n_handshake_io.c

```diff
@@ -813,6 +813,26 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
                 return S2N_SUCCESS;
             }
 
+            if (conn->mode == S2N_SERVER && conn->ems_required && !conn->ems_negotiated && S2N_IN_TEST) {
```

Comment:
These if/else statements are pretty complex. Can they be simplified?  How have you convinced yourself that all possible combinations of these three variables are handled properly?